demo/src/faber.ts

```diff
@@ -0,0 +1,168 @@
+  }
+
+  private printSchema(name: string, version: string, attributes: string[]) {
+    console.log(`\n\nThe credential definition will look like this:\n`)
+    console.log(`${Color.purlpe}Name: ${Color.reset}${name}`)
+    console.log(`${Color.purlpe}Version: ${Color.reset}${version}`)
+    console.log(`${Color.purlpe}Attributes: ${Color.reset}${attributes[0]}, ${attributes[1]}, ${attributes[2]}\n`)
+  }
+
+  private async registerSchema() {
+    const schemaTemplate = {
+      name: 'Faber College' + uuid(),
+      version: '1.0.0',
+      attributes: ['name', 'degree', 'date'],
+    }
+    this.printSchema(schemaTemplate.name, schemaTemplate.version, schemaTemplate.attributes)
+    this.ui.updateBottomBar(`${Color.green}\nRegistering schema...\n`)
+    const schema = await this.agent.ledger.registerSchema(schemaTemplate)
+    this.ui.updateBottomBar('\nSchema registerd!\n')
+    return schema
+  }
+
+  private async registerCredentialDefiniton(schema: Schema) {
+    this.ui.updateBottomBar('\nRegistering credential defenition...\n')
+    this.credentialDefinition = await this.agent.ledger.registerCredentialDefinition({
+      schema,
+      tag: 'latest',
+      supportRevocation: false,
+    })
+    this.ui.updateBottomBar('\nCredential definition registerd!!\n')
+    return this.credentialDefinition
+  }
+
+  private getCredentialPreview() {
+    const credentialPreview = CredentialPreview.fromRecord({
+      name: 'Alice Smith',
+      degree: 'Computer Science',
+      date: '01/01/2022',
+    })
+    return credentialPreview
+  }
+
+  public async issueCredential() {
+    const schema = await this.registerSchema()
+    const credentialDefinition = await this.registerCredentialDefiniton(schema)
+    const credentialPreview = this.getCredentialPreview()
+    const connectionRecord = await this.getConnectionRecord()
+
+    this.ui.updateBottomBar('\nSending credential offer...\n')
+    await this.agent.credentials.offerCredential(connectionRecord.id, {
+      credentialDefinitionId: credentialDefinition.id,
+      preview: credentialPreview,
+    })
+    this.ui.updateBottomBar(`\nCredential offer sent! \n${Color.reset}`)
+  }
+
+  private async printProofFlow(print: string) {
+    this.ui.updateBottomBar(print)
+    await new Promise((f) => setTimeout(f, 2000))
+  }
+
+  private async newProofAttribute() {
+    await this.printProofFlow(`${Color.green}\nCreating new proof attribute for 'name' ...\n`)
+    return {
+      name: new ProofAttributeInfo({
+        name: 'name',
+        restrictions: [
+          new AttributeFilter({
+            credentialDefinitionId: this.credentialDefinition?.id,
+          }),
+        ],
+      }),
+    }
+  }
+
+  public async sendProofRequest() {
+    const connectionRecord = await this.getConnectionRecord()
+    const proofAttribute = await this.newProofAttribute()
+    await this.printProofFlow(`${Color.green}\nRequesting proof...\n`)
```

Comment:
Might be useful to create a util method that takes a color and a message for nicer printing. String interpolation like this everywhere is not very nice.